cmd/rpcdaemon/cli/config.go

```diff
@@ -91,20 +91,30 @@ var rootCmd = &cobra.Command{
 	Short: "rpcdaemon is JSON RPC server that connects to Erigon node for remote DB access",
 }
 
+type PolygonRPCCfg struct {
+	Enabled                      bool
+	StateReceiverContractAddress string
+}
+
 var (
 	stateCacheStr string
 )
 
-func RootCommand() (*cobra.Command, *httpcfg.HttpCfg) {
+func RootCommand() (*cobra.Command, *httpcfg.HttpCfg, *PolygonRPCCfg) {
 	utils.CobraFlags(rootCmd, debug.Flags, utils.MetricFlags, logging.Flags)
 
 	cfg := &httpcfg.HttpCfg{Sync: ethconfig.Defaults.Sync, Enabled: true, StateCache: kvcache.DefaultCoherentConfig}
+	polygonSyncCfg := &PolygonRPCCfg{Enabled: false}
+
 	rootCmd.PersistentFlags().StringVar(&cfg.PrivateApiAddr, "private.api.addr", "127.0.0.1:9090", "Erigon's components (txpool, rpcdaemon, sentry, downloader, ...) can be deployed as independent Processes on same/another server. Then components will connect to erigon by this internal grpc API. Example: 127.0.0.1:9090")
 	rootCmd.PersistentFlags().StringVar(&cfg.DataDir, "datadir", "", "path to Erigon working directory")
 	rootCmd.PersistentFlags().BoolVar(&cfg.GraphQLEnabled, "graphql", false, "enables graphql endpoint (disabled by default)")
 	rootCmd.PersistentFlags().Uint64Var(&cfg.Gascap, "rpc.gascap", 50_000_000, "Sets a cap on gas that can be used in eth_call/estimateGas")
 	rootCmd.PersistentFlags().Uint64Var(&cfg.MaxTraces, "trace.maxtraces", 200, "Sets a limit on traces that can be returned in trace_filter")
 
+	rootCmd.PersistentFlags().BoolVar(&polygonSyncCfg.Enabled, "polygon.sync", false, "Enable if Erigon has been synced using the new polygon sync component")
+	rootCmd.PersistentFlags().StringVar(&polygonSyncCfg.StateReceiverContractAddress, "polygon.sync.stateReceiverContractAddress", "0x0000000000000000000000000000000000001001", "State receiver contract for polygon sync component")
```

Comment:
We do not pass this via input flags.

We fetch the chain config from the DB here - https://github.com/erigontech/erigon/blob/main/cmd/rpcdaemon/cli/config.go#L376

You need to use that and instantiate the bridge in RemoteServices where the ChainConfig is available for the WithDatadir case.